### PR TITLE
EPMRPP-38332 || GitHub oauth refactoring

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/reportportal/auth/OAuthSuccessHandler.java [hunk 1]
```diff
@@ -65,19 +65,17 @@
 
 	@Override
 	protected void handle(HttpServletRequest request, HttpServletResponse response, Authentication authentication) throws IOException {
-		String login = ofNullable(((OAuth2User) authentication.getPrincipal()).getAttributes().get(LOGIN_ATTRIBUTE)).map(String::valueOf)
+		String login = ofNullable((OAuth2Authentication) authentication).map(auth -> String.valueOf(auth.getPrincipal()))
 				.orElseThrow(() -> new ReportPortalException(ErrorType.ACCESS_DENIED,
-						Suppliers.formattedSupplier("Attribute - {} was not provided.", LOGIN_ATTRIBUTE).get()
+						Suppliers.formattedSupplier("{} was not provided.", LOGIN_ATTRIBUTE).get()
```
Comment (ihar-kahadouski) [inline src/main/java/com/epam/reportportal/auth/OAuthSuccessHandler.java]: It seems we dont need login_attribute constatnt anymore

File: src/main/java/com/epam/reportportal/auth/integration/github/GitHubUserReplicator.java [hunk 2]
```diff
@@ -108,63 +120,62 @@ public User replicateUser(String accessToken) {
...
+				throw new UserSynchronizationException("User with login '" + u.getId() + "' already exists");
 			}
+			return u;
+		}).orElseGet(() -> userRepository.save(createUser(userResource, gitHubClient)));
+	}
+
+	private void updateUser(User user, UserResource userResource, GitHubClient gitHubClient) {
+		String email = userResource.getEmail();
+		if (Strings.isNullOrEmpty(email)) {
+			email = retrieveEmail(gitHubClient).orElseThrow(() -> new UserSynchronizationException("User 'email' has not been provided"));
```
Comment (ihar-kahadouski) [inline src/main/java/com/epam/reportportal/auth/integration/github/GitHubUserReplicator.java]: There is a lot of duplicate code. Can you refactor it?

File: src/main/java/com/epam/reportportal/auth/OAuthSuccessHandler.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/auth/OAuthSuccessHandler.java]: removed

File: src/main/java/com/epam/reportportal/auth/integration/github/GitHubUserReplicator.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/auth/integration/github/GitHubUserReplicator.java]: updated

File: src/main/java/com/epam/reportportal/auth/AuthConfigService.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2016 EPAM Systems
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/auth/AuthConfigService.java]: Apache 2.0 license

File: src/main/java/com/epam/reportportal/auth/event/UiUserSignedInEvent.java [hunk 4]
```diff
@@ -35,6 +35,7 @@
 
 	public UiUserSignedInEvent(Authentication authentication) {
 		super(authentication);
+		System.err.println("HALLLOOOO");
```
Comment (ihar-kahadouski) [inline src/main/java/com/epam/reportportal/auth/event/UiUserSignedInEvent.java]: Remove it

File: src/main/java/com/epam/reportportal/auth/OAuthSuccessHandler.java [hunk 5]
```diff
@@ -59,31 +60,39 @@
...
+						ofNullable(oAuth2Authentication.getOAuth2Request()).map(OAuth2Request::getExtensions)
+								.orElseGet(Collections::emptyMap)
+				);
 
 		MultiValueMap<String, String> query = new LinkedMultiValueMap<>();
 		query.add("token", accessToken.getValue());
 		query.add("token_type", accessToken.getTokenType());
-		//TODO replace hard-coded port with custom value
 		URI rqUrl = UriComponentsBuilder.fromHttpRequest(new ServletServerHttpRequest(request))
-				.port(8080)
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/auth/OAuthSuccessHandler.java]: so what port is used here now?

File: src/main/java/com/epam/reportportal/auth/event/UiUserSignedInEvent.java (hunk 4 above)
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/auth/event/UiUserSignedInEvent.java]: ?

File: src/main/java/com/epam/reportportal/auth/integration/github/ExternalOauth2TokenConverter.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
+package com.epam.reportportal.auth.integration.github;
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/auth/integration/github/ExternalOauth2TokenConverter.java]: Header

File: src/main/java/com/epam/reportportal/auth/integration/github/GitHubTokenServices.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
+/*
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/auth/integration/github/GitHubTokenServices.java]: Apache 2.0

File: src/main/java/com/epam/reportportal/auth/OAuthSuccessHandler.java (hunk 5 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/auth/OAuthSuccessHandler.java]: Port from where request came from

File: src/main/java/com/epam/reportportal/auth/event/UiUserSignedInEvent.java (hunk 4 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/auth/event/UiUserSignedInEvent.java]: fixed

Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/auth/event/UiUserSignedInEvent.java]: removed

File: src/main/java/com/epam/reportportal/auth/AuthConfigService.java (hunk 3 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/auth/AuthConfigService.java]: fixed

File: src/main/java/com/epam/reportportal/auth/integration/github/ExternalOauth2TokenConverter.java (hunk 6 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/auth/integration/github/ExternalOauth2TokenConverter.java]: added

File: src/main/java/com/epam/reportportal/auth/integration/github/GitHubTokenServices.java (hunk 7 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/auth/integration/github/GitHubTokenServices.java]: added

**Pull request merged**